fix(.NET): Release creation (#10134)
getsentry/sentry-docs: platform-includes/set-release/dotnet.mdx

The SDK attempts to locate the release to add to events sent to Sentry.

The SDK will first check for a version set on the environment via `SENTRY_RELEASE` and use it as-is.

> If no version can be found it falls back to look at the [entry assembly's](<https://msdn.microsoft.com/en-us/library/system.reflection.assembly.getentryassembly(v=vs.110).aspx>) `AssemblyInformationalVersionAttribute`, which accepts a string as value and is often used to set a GIT commit hash. If that returns null, it'll look at the default `AssemblyVersionAttribute` which accepts the numeric version number. The resulting release will be in the format `assembly-name@version`.

Reviewer: ```suggestion
If no version is found, the SDK will look at the [entry assembly's](<https://msdn.microsoft.com/en-us/library/system.reflection.assembly.getentryassembly(v=vs.110).aspx>) `AssemblyInformationalVersionAttribute`, which accepts a string value and is often used to set a GIT commit hash. If that returns null, then the SDK will look at the default `AssemblyVersionAttribute`, which accepts the numeric version number. The resulting release will be in the format `<assembly-name>@<version-number>`.
```
